### PR TITLE
Enhancement to previous PR

### PR DESCRIPTION
**Pull request by pavitra14**
This enhances the previous pull request, https://github.com/byliuyang/short-ext/pullshttps://github.com/byliuyang/short-ext/pull/12.

## Current Behavior ( Optional for new feature )
Nothing happens when user clicks on extension icon in Chrome browser.

## New Behavior
Whenever user clicks on the extension, it'll automatically grab the url and open a new tab with the shortening service, allowing users to use the service with literally 1 click

### CODE REVIEW

File: src/background.ts [hunk 1]
```diff
@@ -34,6 +34,8 @@ function openTab(url: string) {
 class ShortExt {
     constructor(private apiBaseUrl: string, private webUi: string) {
         this.setupOmnibox();
+        // Call the Event Listener Function bind onclick
```
Comment (naturetime10) [inline src/background.ts]:
```suggestion
        // Execute when extension icon is clicked
```

File: src/background.ts [hunk 2]
```diff
@@ -91,6 +93,22 @@ class ShortExt {
     launch() {
         this.interceptRequests();
     }
+
+    // Function to handle extension clicks
```
Comment (naturetime10) [inline src/background.ts]:
```suggestion
```

This comment doesn't provide much information.

File: src/background.ts [hunk 3]
```diff
@@ -91,6 +93,22 @@ class ShortExt {
...
+    // Required: browser_action clause in manifest.json to tell the browser that we will need it's events.
```
Comment (naturetime10) [inline src/background.ts]:
```suggestion
```

File: src/background.ts [hunk 4]
```diff
@@ -91,6 +93,22 @@ class ShortExt {
...
+    redirectToBaseURI = (base_uri: String) => {
+        // Event Listener
```
Comment (naturetime10) [inline src/background.ts]:
```suggestion
      //  browser_action need to be configured in manifest.json
```

File: src/background.ts [hunk 5]
```diff
@@ -91,6 +93,22 @@ class ShortExt {
...
+        chrome.browserAction.onClicked.addListener(function(tab)
```
Comment (naturetime10) [inline src/background.ts]:
```suggestion
        chrome
          .browserAction
          .onClicked
          .addListener(tab => 
```

File: src/background.ts [hunk 6]
```diff
@@ -91,6 +93,22 @@ class ShortExt {
...
     }
+
+    // Function to handle extension clicks
+    // Required: browser_action clause in manifest.json to tell the browser that we will need it's events.
+    redirectToBaseURI = (base_uri: String) => {
+        // Event Listener
+        chrome.browserAction.onClicked.addListener(function(tab) 
+            { 
+                // Check if the current tab has some url/event should not be triggered on empty tabs
+                if(tab.hasOwnProperty("url")){
```
Comment (naturetime10) [inline src/background.ts]:
```suggestion
                if(isEmptyTab(tab)) {
                  return;
                }
```

File: src/background.ts [hunk 7]
```diff
@@ -91,6 +93,22 @@ class ShortExt {
...
+                    url = tab.url;
+                    chrome
+                    .tabs
+                    .create({ url: `${base_uri}/?long_link=${url}` }, () => {});
+                }
```
Comment (naturetime10) [inline src/background.ts]:
```suggestion
```

File: src/background.ts (hunk 6 above)
Comment (naturetime10) [reply src/background.ts]:
```typescript
isEmptyTab(tab) {
  return !tab.hasOwnProperty("url");
}
```

Please add `isEmptyTab` method after the refactor.
In general, we prefer readable variable & function names over comments. Comments are added when there are hidden knowledge that looks confusing to the readers.

**Review by naturetime10 — CHANGES_REQUESTED** [review]

File: src/background.ts [hunk 8]
```diff
@@ -91,6 +93,22 @@ class ShortExt {
     launch() {
         this.interceptRequests();
     }
+
+    // Function to handle extension clicks
+    // Required: browser_action clause in manifest.json to tell the browser that we will need it's events.
+    redirectToBaseURI = (base_uri: String) => {
```
Comment (naturetime10) [inline src/background.ts]:
```suggestion
    redirectToHomePage = (homepageURL: String) => {
```

File: src/background.ts [hunk 9]
```diff
@@ -34,6 +34,8 @@ function openTab(url: string) {
 class ShortExt {
     constructor(private apiBaseUrl: string, private webUi: string) {
         this.setupOmnibox();
+        // Call the Event Listener Function bind onclick
+        this.redirectToBaseURI(webUi);
```
Comment (naturetime10) [inline src/background.ts]:
```suggestion
        this.redirectToHomePage(webUi);
```

File: src/background.ts [hunk 10]
```diff
@@ -91,6 +93,22 @@ class ShortExt {
...
+    redirectToBaseURI = (base_uri: String) => {
+        // Event Listener
+        chrome.browserAction.onClicked.addListener(function(tab) 
+            { 
+                // Check if the current tab has some url/event should not be triggered on empty tabs
+                if(tab.hasOwnProperty("url")){
+                    url = tab.url;
+                    chrome
+                    .tabs
+                    .create({ url: `${base_uri}/?long_link=${url}` }, () => {});
```
Comment (naturetime10) [inline src/background.ts]:
```suggestion
                    .create({ url: `${homepageURL}/?long_link=${currentPageURL}` }, () => {});
```

File: src/background.ts [hunk 11]
```diff
@@ -91,6 +93,22 @@ class ShortExt {
...
+
+    // Function to handle extension clicks
+    // Required: browser_action clause in manifest.json to tell the browser that we will need it's events.
+    redirectToBaseURI = (base_uri: String) => {
+        // Event Listener
+        chrome.browserAction.onClicked.addListener(function(tab) 
+            { 
+                // Check if the current tab has some url/event should not be triggered on empty tabs
+                if(tab.hasOwnProperty("url")){
+                    url = tab.url;
```
Comment (naturetime10) [inline src/background.ts]:
```suggestion
                    currentPageURL = tab.url;
```

File: src/background.ts (hunk 10 above)
Comment (naturetime10) [reply src/background.ts]: Do we need `() => {}` for the second parameter?

**Review by naturetime10 — CHANGES_REQUESTED** [review]

File: src/background.ts (hunk 10 above)
Comment (pavitra14) [reply src/background.ts]: It's an optional callback for create method, might be useful if there is a need to get that tab's instance.
Will remove for now.

File: src/background.ts [hunk 12]
```diff
@@ -91,6 +93,30 @@ class ShortExt {
...
     }
+
+    redirectToHomePage = (homepageURL: String) => {
+      //  browser_action need to be configured in manifest.json
+        chrome
+          .browserAction
+          .onClicked
+          .addListener(tab => 
+            { 
+                // Check if the current tab has some url/event should not be triggered on empty tabs
```
Comment (naturetime10) [inline src/background.ts]:
```suggestion
```
This comment can be removed. This code is very self-explanatory already.

**Review by naturetime10 — CHANGES_REQUESTED** [review]

**Comment by pavitra14** [comment]
Kindly check.

**Review by naturetime10 — APPROVED** [review]

**Pull request merged**